src/pb_stub.cc

```diff
@@ -1032,9 +1032,9 @@ Stub::~Stub()
 
   {
     py::gil_scoped_acquire acquire;
-    async_event_loop_ = py::none();
-    background_futures_ = py::none();
-    model_instance_ = py::none();
+    py::object async_event_loop_local(std::move(async_event_loop_));
+    py::object background_futures_local(std::move(background_futures_));
+    py::object model_instance_local(std::move(model_instance_));
```

Comment:
Just for my own understanding - does this mean that `async_event_loop_ = py::none();` will not actually delete the python object?

Reply:
It may not. When I tried the set to none method in the tensor destructor, the object was not destructed.

Reply:
You have to decrement the reference of the object for it to be garbage collected. Once the reference count of the object goes to zero, then the object will be picked for garbage collection. My guess is that assigning these variables to `py::none()` didn't decrement the reference count on these objects if you were to inspect the properties at runtime again. I think this idea is also reinforced since you use the `std::move` operator which does reference counting on the smart pointers being used. All this should be taken with a caveat that I haven't worked in this part of the code and is only based on this PR and your conversations.